cmake/modules/AddCephTest.cmake

```diff
@@ -17,10 +23,19 @@ function(add_ceph_test test_name test_path)
     PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY}:${CMAKE_SOURCE_DIR}/src:$ENV{PATH}
     PYTHONPATH=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cython_modules/lib.${Python${PYTHON_VERSION}_VERSION_MAJOR}:${CMAKE_SOURCE_DIR}/src/pybind
     CEPH_BUILD_VIRTUALENV=${CEPH_BUILD_VIRTUALENV})
-  # none of the tests should take more than 1 hour to complete
-  set_property(TEST
-    ${test_name}
-    PROPERTY TIMEOUT ${CEPH_TEST_TIMEOUT})
+  if(DEFINED PROPS_TIMEOUT)
+    set_property(TEST ${test_name} PROPERTY TIMEOUT ${PROPS_TIMEOUT})
+  else()
```

Comment:
Any chance to have a shared macro doing this (set label, cost and timeout) for both `add_ceph_test` and `add_tox_test`?

Reply:
yeah, when I plugged it in for the 3rd time, I got sort of wondering.
You think I should?!

Reply:
I'd say so. That chunk (setting LABELS, COST and TIMEOUT) could/should be common to the 3 kind of tests. If we need to add some specific label (like unittest/tox) we may just to a `list(APPEND)` with that.

Reply:
@epuertat 
Created a marco to do the grunt work.
Wonderfull difference between function and macro.